### DISCUSSION TITLE
PDF Viewer, sidebar/column resizing

### DISCUSSION THREAD

**Discussion by mjadobson** [pocketbase/pocketbase#1542] [Q&A]
I've made a couple of small tweaks to the admin UI and was wondering if you would accept pull requests for them.

The image popup has been expanded to include PDF embedding:

<img width="1440" alt="Screenshot 2023-01-08 at 22 57 11" src="https://user-images.githubusercontent.com/787996/211221116-fc881d73-b85a-4d6b-b8f9-a36920cf195d.png">

The sidebar and collection columns are resizeable:

![chrome-capture-2023-0-8](https://user-images.githubusercontent.com/787996/211221132-088d01fc-b761-425e-bc72-a7fd77daa83f.gif)

Thanks again for your work 👍

**Comment by ganigeorgiev** [comment]
I don't think the collections sidebar resizing is necessary and it may complicate things in the future when we decide to make the dashboard responsive/mobile friendly.

For the PDF preview - I don't know whether it would be very useful since you can get a full preview by directly clicking on the file. Additionally it will not be consistent with the regular files since it will be the only clickable placeholder without thumb image.
I guess we can add it if it is using a native browser rendering (_eg. embed/iframe; I don't know what is the compatibility and whether it works everywhere but it should be better than complex js plugins that may fail with large files_) but also maybe attach a link to the other non-previewable files to open in a new tab so that all file placeholders could have an action.

**Comment by mjadobson** [comment]
Thanks for the feedback, I'll organise a pull request for PDF previewing and see what you think.

My gif didn't include it, but it also includes resizing collection columns as well as the sidebar. I find resizing columns useful as some of my collections are data heavy and it helps bring relevant columns into a single view without horizontal scrolling. Pocketbase is quite generous with its padding/spacing as well, but my preference is for a more compact UI for table views (like with airtable/baserow/excel/etc) where I get a better overview of the data.

**Reply by ganigeorgiev** [reply]
I'm not sure how fixed resizable columns will work when combined with the columns visibility (from the 3 dots at in the right corner). I think it will complicate too much the listing implementation but I may be wrong, I don't know. If you have already implemented it, feel free to file it in a separate PR and I'll test it.

**Comment by dschissler** [comment]
@mjadobson Are you embedding pdf.js into Pocketbase?  If so, then that will add a lot of bloat to the binary.  If it's using native browser PDF rendering then that would be better.

**Reply by mjadobson** [reply]
It uses native embedding. In my experience this works well for all main desktop browsers, but iOS/android would require pulling in pdf.js
